code/game/objects/items/devices/PDA/PDA.dm

```diff
@@ -1449,3 +1443,91 @@ var/global/list/obj/item/device/pda/PDAs = list()
 /obj/item/device/pda/emp_act(severity)
 	for(var/atom/A in src)
 		A.emp_act(severity)
+
+
+
+//Light functions, to break stacking and support diona checking
+//===============================================================
+/obj/item/device/pda/var/mob/living/carbon/holder //The person who's holding this PDA
+/obj/item/device/pda/var/exploit_protection = 1 //Used to prevent light stacking
+/obj/item/device/pda/var/emitting = 0 //Used to prevent unnecessarily setting the light, premature optimisation to prevent lag
+
+
+/obj/item/device/pda/var/fon = 0 //Is the flashlight function on?
+/obj/item/device/pda/var/f_lum = 2 //Luminosity for the flashlight function
+
+/obj/item/device/pda/proc/turn_on()

```

Comment:
Please clarify the proc name here to mean the actual light. The PDA can be "turned on" in multiple ways.